lib/samples/show_device_location_with_nmea_data_sources/show_device_location_with_nmea_data_sources.dart

```diff
@@ -0,0 +1,300 @@
+                              // Set the autoPanMode to recenter.
+                              _mapViewController.locationDisplay.autoPanMode =
+                                  LocationDisplayAutoPanMode.recenter;
+                            }
+                          : null,
+                      child: const Text('Recenter'),
+                    ),
+                    // Stop and reset the location data source.
+                    ElevatedButton(
+                      onPressed:
+                          _locationDataSourceRunning ? _stopDataSource : null,
+                      child: const Text('Reset'),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+            // Widget to show top details.
+            NmeaLocationDetails(
+              nmeaLocation: _currentNmeaLocation,
+              nmeaSatelliteInfos: _currentSatelliteInfos,
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> onMapViewReady() async {
+    // Create a map and set it to the MapView.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
+    _mapViewController.arcGISMap = map;
+
+    // Use the NMEA location data source as the data source for the map.
+    _mapViewController.locationDisplay.dataSource = _locationDataSource;
+
+    // Set the autoPanMode to recenter and listen for any changes.
+    _mapViewController.locationDisplay.autoPanMode =
+        LocationDisplayAutoPanMode.recenter;
+    _autopanSubscription = _mapViewController
+        .locationDisplay.onAutoPanModeChanged
+        .listen((autoPanMode) {
+      setState(() {
+        // Activates/deactivates the Recenter button based on the new auto pan mode.
+        _enableRecenter = autoPanMode != LocationDisplayAutoPanMode.recenter;
+      });
+    });
+
+    // Set the ready state variable to true to enable the UI.
+    setState(() => _ready = true);
+  }
+
+  Future<void> _startDataSource() async {
+    // Create new instance of the NmeaSourceSimulator.
+    _nmeaDataSimulator ??= SimulatedNmeaDataSource();
+
+    // Subscribe to the simulator data.
+    _nmeaDataSubscription ??=
+        _nmeaDataSimulator!.nmeaMessages.listen((nmeaDataString) {
+      final nmeaData = utf8.encoder.convert(nmeaDataString);
+      _locationDataSource.pushData(nmeaData);
+    });
+
+    // Subscribe to the location stream of the location data source.
+    _locationSubscription ??=
+        _locationDataSource.onLocationChanged.listen((location) {
+      setState(() => _currentNmeaLocation = location);
+    });
+
+    // Subscribe to the location data source's satellite changed stream.
+    _satelliteSubscription ??=
+        _locationDataSource.onSatellitesChanged.listen((satelliteInfos) {
+      setState(() => _currentSatelliteInfos = satelliteInfos);
+    });
```

Comment:
I'm not so sure about all these `??=`.

`_locationSubscription` and `_satelliteSubscription` could be set up just once in `onMapViewReady` at the same time it does `_mapViewController.locationDisplay.dataSource = _locationDataSource`.